### PR TITLE
Cleanup Makefile

### PR DESCRIPTION
**Pull request by kron4eg**
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,86 +14,69 @@
...
 
-e2e-test: dist/kubeone
-	CGO_ENABLED=1 ./hack/run-ci-e2e-test.sh
+.PHONY: e2e-test
+e2e-test: install
+	./hack/run-ci-e2e-test.sh
 
+.PHONY: buildenv
+buildenv:
+	@go version
```
Comment (kron4eg) [inline Makefile]: This line reports: `go version go1.11.4 linux/amd64`

File: Makefile [hunk 2]
```diff
@@ -14,86 +14,69 @@
...
-verify-dependencies:
+.PHONY: verify-dependencies
+verify-dependencies: buildenv
 	go mod verify
 
-.PHONY: test e2e-test lint verify-licence verify-codegen verify-boilerplate
+.PHONY: test
 test:
-	CGO_ENABLED=1 go test $(GOBUILDFLAGS) -race ./pkg/... ./test/...
+	CGO_ENABLED=1 go test -race ./pkg/... ./test/...
```
Comment (xmudrii) [inline Makefile]: Why do we enable cgo here?

Comment (kron4eg) [reply Makefile]: Because of -race requesting it

**Pull request merged**